extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java

```diff
@@ -13,6 +13,10 @@
 
     public static final String HTTP = "http://";
     public static final String HTTPS = "https://";
+    /**
+     * @deprecated Use {@link java.nio.charset.StandardCharsets#UTF_8} instead
```

Comment:
This is required for KitKat support. KitKat does not know `StandardCharset`.

Reply:
Since Kitkat support is being removed, shouldn't this be okay now?

Reply:
> This is required for KitKat support. KitKat

Uhm what?
We are compiling against the Java and not Android.

Btw this was added in API 19 aka ``KitKat`` according to the docs so no problems here I guess 😄 
https://developer.android.com/reference/java/nio/charset/StandardCharsets

Reply:
Ah sorry, for the legacy version, so < KitKat

Reply:
> Ah sorry, for the legacy version, so < KitKat

Okay so let me clarify this: We can't use all Java 8 features - against which we are compiling - because we have to support an outdated > 9 year old Android version. And some person needs to tell you that otherwise the Extractor won't work anymore (on certain devices).
I wonder how this did even work so far and didn't result in major problems.
We should be definitely rework this. (Maybe by dropping the NP-Legacy - which is currently unmaintained).

I will revert the changes for now but this has to change.

Reply:
Is it not possible to have version specific code here? The `build.sdk.kitkat` thing.

Reply:
Doesn't desugaring magically solve this issue?

Reply:
> Is it not possible to have version specific code here? The build.sdk.kitkat thing.

NewPipeExtractor is built on Java not Android. So no.

> Doesn't desugaring magically solve this issue?

Not sure about that one but I couldn't find anything inside https://developer.android.com/studio/write/java8-support-table.